src/vs/workbench/contrib/comments/browser/commentsEditorContribution.ts

```diff
@@ -495,28 +496,32 @@ export class CommentController implements IEditorContribution {
 			});
 
 			changed.forEach(thread => {
-				let matchedZones = this._commentWidgets.filter(zoneWidget => zoneWidget.owner === e.owner && zoneWidget.commentThread.threadId === thread.threadId);
-				if (matchedZones.length) {
-					let matchedZone = matchedZones[0];
-					matchedZone.update(thread);
+				if (thread.isDocumentCommentThread()) {
```

Comment:
We now need to have these checks in various places. Would it make more sense to have a separate notebook comments service?

Reply:
or, at the very least, separate `onDidUpdateCommentThreads` events for documents and notebooks?

Reply:
💯 for filtering through `onDidUpdateCommentThreads`.